File: IntValue.hs
{-# LANGUAGE TypeFamilies, DataKinds #-}

module IntValue where

import Java
import ExpValue

data IntValue = IntValue @com.eunmin.IntValue deriving Class

foreign import java unsafe "@new" newIntValue :: Int -> Java a IntValue

foreign import java unsafe getValue :: Java IntValue Bool

type instance Inherits IntValue = '[ExpValue]

File: java/com/eunmin/IntValue.java
package com.eunmin;

public class IntValue implements ExpValue {
  private Integer value;

  public IntValue(int value) {
      this.value = value;
  }

  public int getValue() {
    return value;
  }

  @Override
  public String getName() {
    return "IntValue";
  }
}
